jira.go

```diff
@@ -63,46 +63,59 @@ func getJiraTickets(endpointAPI string, orgID string, projectID string, token st
 
-		jsonVuln, _ := jsn.NewJson(vulnForJira)
-		vulnID := jsonVuln.K("id").String().Value
-		jiraTicket := formatJiraTicket(jsonVuln, projectInfo)
+	jsonVuln, _ := jsn.NewJson(vulnForJira)
+	vulnID := jsonVuln.K("id").String().Value
+	jiraTicket := formatJiraTicket(jsonVuln, projectInfo)
 
-		jiraTicket.Fields.Projects.ID = jiraProjectID
-		jiraTicket.Fields.IssueTypes.Name = jiraTicketType
-		jiraTicket.Fields.Assignees.ID = assigneeID
-		if labels != "" {
-			jiraTicket.Fields.Labels = strings.Split(labels, ",")
-		}
-
-		if priorityIsSeverity {
-			var priority PriorityType
-			jiraMappingEnvVarName := fmt.Sprintf("SNYK_JIRA_PRIORITY_FOR_%s_VULN", strings.ToUpper(jsonVuln.K("severity").String().Value))
-			val, present := os.LookupEnv(jiraMappingEnvVarName)
-			if present {
-				priority.Name = val
+	jiraTicket.Fields.Projects.ID = jiraProjectID
+	jiraTicket.Fields.IssueTypes.Name = jiraTicketType
+	jiraTicket.Fields.Assignees.ID = assigneeID
+	if labels != "" {
+		jiraTicket.Fields.Labels = strings.Split(labels, ",")
+	}
+	if priorityIsSeverity {
+		var priority PriorityType
+		jiraMappingEnvVarName := fmt.Sprintf("SNYK_JIRA_PRIORITY_FOR_%s_VULN", strings.ToUpper(jsonVuln.K("severity").String().Value))
+		val, present := os.LookupEnv(jiraMappingEnvVarName)
+		if present {
+			priority.Name = val
+		} else {
+			if jsonVuln.K("severity").String().Value == "critical" {
+				priority.Name = "Highest"
 			} else {
-				if jsonVuln.K("severity").String().Value == "critical" {
-					priority.Name = "Highest"
-				} else {
 
-					priority.Name = strings.Title(jsonVuln.K("severity").String().Value)
+				priority.Name = strings.Title(jsonVuln.K("severity").String().Value)
 
-				}
 			}
-			jiraTicket.Fields.Priority = &priority
-		}
 
-		ticket, err := json.Marshal(jiraTicket)
-		if err != nil {
-			log.Fatalln(err)
 		}
+		jiraTicket.Fields.Priority = &priority
+	}
 
-		responseData := makeSnykAPIRequest("POST", endpointAPI+"/v1/org/"+orgID+"/project/"+projectInfo.K("id").String().Value+"/issue/"+vulnID+"/jira-issue", token, ticket)
-		responseDataAggregated += "\n" + string(responseData)
+	ticket, err := json.Marshal(jiraTicket)
+	if err != nil {
+		log.Fatalln(err)
 	}
-	return responseDataAggregated
+
+	responseData := makeSnykAPIRequest("POST", endpointAPI+"/v1/org/"+orgID+"/project/"+projectInfo.K("id").String().Value+"/issue/"+vulnID+"/jira-issue", token, ticket)
+
+	return responseData
+}
+
+func openJiraTickets(endpointAPI string, orgID string, token string, jiraProjectID string, jiraTicketType string, assigneeID string, labels string, projectInfo jsn.Json, vulnsForJira map[string]interface{}, priorityIsSeverity bool) string {
+	fullResponseDataAggregated := ""
+
+	for _, vulnForJira := range vulnsForJira {
+		responseDataAggregatedByte := openJiraTicket(endpointAPI, orgID, token, jiraProjectID, jiraTicketType, assigneeID, labels, projectInfo, vulnForJira, priorityIsSeverity)
```

Comment:
probably cleaner to have openJiraTicket return responseDataAggregatedByte, err := and use if err != nil then retry opening a ticket again. Go has this nice multi member return value thingy that is nice to use for this sort of things. It's cleaner than parsing the response. And if parsing the response is the only way, then keep it into openjiraticket logic so we contain the opening logic inside the function.

Reply:
I pushed some changes. But I want to keep openJiraTicket and openJiraTickets separated as I need to call openJiraTicket twice.

Reply:
Yes it's fine, the important bit is that openjiraticket handles the ticket opening and deals with the error interpretation. That needs to stay inside this function, so the change is good